sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -0,0 +1,892 @@
+        """
+        Provide an approximate representative point for a given cluster.
+
+        Note that this technique can be very slow and memory intensive for
+        large clusters. For faster results use the `weighted_cluster_centroid`
+        method which is faster, but assumes a euclidean metric.
+
+        Parameters
+        ----------
+        cluster_id : int
+            The id of the cluster to compute a medoid for.
+
+        Returns
+        -------
+        centroid : array of shape (n_features,)
+            A representative medoid for cluster `cluster_id`.
+        """
+        if not hasattr(self, "labels_"):
+            raise AttributeError("Model has not been fit to data")
+
+        if cluster_id == -1:
+            raise ValueError(
+                "Cannot calculate weighted centroid for -1 cluster "
+                "since it is a noise cluster"
+            )
+
+        mask = self.labels_ == cluster_id
+        cluster_data = self._raw_data[mask]
+        cluster_membership_strengths = self.probabilities_[mask]
+        metric_params = self.metric_params or {}
+
+        dist_mat = pairwise_distances(cluster_data, metric=self.metric, **metric_params)
+
+        dist_mat = dist_mat * cluster_membership_strengths
+        medoid_index = np.argmin(dist_mat.sum(axis=1))
+        return cluster_data[medoid_index]
+
+    def dbscan_clustering(self, cut_distance, min_cluster_size=5):
+        """
+        Return clustering given by DBSCAN without border points.
+
+        Return clustering that would be equivalent to running DBSCAN* for a
+        particular cut_distance (or epsilon) DBSCAN* can be thought of as
+        DBSCAN without the border points.  As such these results may differ
+        slightly from `cluster.DBSCAN` due to the difference in implementation
+        over the non-core points.
+
+        This can also be thought of as a flat clustering derived from constant
+        height cut through the single linkage tree.
+
+        This represents the result of selecting a cut value for robust single linkage
+        clustering. The `min_cluster_size` allows the flat clustering to declare noise
+        points (and cluster smaller than `min_cluster_size`).
+
+        Parameters
+        ----------
+
+        cut_distance : float
+            The mutual reachability distance cut value to use to generate a
+            flat clustering.
+
+        min_cluster_size : int, optional
+            Clusters smaller than this value with be called 'noise' and remain
+            unclustered in the resulting flat clustering.
+
+        Returns
+        -------
+
+        labels : array [n_samples]
+            An array of cluster labels, one per datapoint. Unclustered points
+            are assigned the label -1.
+        """
+        return labelling_at_cut(
+            self._single_linkage_tree_, cut_distance, min_cluster_size
+        )
+
+    def _more_tags(self):
+        return {"allow_nan": True}
```

Comment:
Given https://github.com/scikit-learn/scikit-learn/pull/22616/files#r959838671, I think this is conditioned on `self.metric == "precomputed"`.